[docs][joy-ui] Revise the Drawer page (#38988)
mui/material-ui: docs/data/joy/components/drawer/drawer.md

---
productId: joy-ui
title: React Drawer component

> components: Drawer, ModalClose, DialogContent

Reviewer: @danilo-leal GH wouldn't let me comment on these lines up here because they weren't part of the PR before, so I went ahead and pushed this change so you can see how to add component links to the API section! 🤝

Author: Sweet, that's perfect! I didn't know it was that simple 😅

# Drawer

<p class="description">Navigation drawers provide quick access to other app areas without removing the user out of context.</p>

## Introduction

The Drawer component is commonly used as a type of menu on desktop and as a dialog on mobile (similar to [Apple's sheets](https://developer.apple.com/design/human-interface-guidelines/sheets)).

{{"demo": "DrawerUsage.js", "hideToolbar": true, "bg": "gradient"}}

## Basic

The Drawer can be closed by clicking on the overlay or pressing the <kbd>Esc</kbd> key.
It will also close if an item inside of it is selected, which can be controlled by the `open` prop.

{{"demo": "DrawerBasic.js"}}

> ## Customization

Author: @siriwatknp + @samuelsycamore should we say something / add a demo about modal vs. non-modal Drawers? I've seen plenty of use cases for both out there in the wild.

Author: @mnajdova Is there anything specific we should do to get a non-modal Drawer? (Image Notion's comment drawer as an example of it).

> ### Anchor

Author: @siriwatknp It seems the default anchor value is `left`, right? Shouldn't we mention it as well as add a Playground control for it?

Reviewer 2: Yes, we should mention it. We can move the toggle button to an anchor in the playground and have just one button.

Author: Agreed, yeah! That'd communicate what is the default one and allow visitors to see what other positions look like!
Let me know if I can help with anything as far as doing it!

Author: @siriwatknp Can you help out with this one? 😬

Author: Thank you! 🎉 🙌

Use the anchor prop to specify where the drawer should appear from.

{{"demo": "DrawerAnchor.js"}}

### Close button

> Use the Modal Close component to add a close button to the Drawer.

Author: @siriwatknp Any particular reason for using this Modal Close as opposed to just a button? What does it bring, specifically? Maybe we make this clearer.

Reviewer 2: Since the Drawer is built on top of the Modal, ModalClose can be used. We could create a DrawerClose component but it will be almost the same as the ModalClose.

Reviewer 2: Using the ModalClose, you get the close functionality by default without providing `onClick` to the button.

```js
<Drawer>
  <ModalClose /> // able to close the drawer.
</Drawer>
```

{{"demo": "DrawerCloseButton.js"}}

### Size

> Use the `size` prop to adjust the Drawer's width.

Author: @siriwatknp Is it accurate to say that this prop only controls the _width_? Because when it's anchored to the top or bottom, it should instead control the _height,_ right?

Reviewer 2: Yes, it controls the `width` or `height` based on the anchor.

# Drawer

<p class="description">Navigation drawers provide quick access to other app areas without removing the user out of context.</p>

## Introduction

The Drawer component is commonly used as a type of menu on desktop and as a dialog on mobile (similar to [Apple's sheets](https://developer.apple.com/design/human-interface-guidelines/sheets)).

{{"demo": "DrawerUsage.js", "hideToolbar": true, "bg": "gradient"}}

## Basic

The Drawer can be closed by clicking on the overlay or pressing the <kbd>Esc</kbd> key.

> It will also close if an item inside of it is selected, which can be controlled by the `open` prop.

Reviewer: To be more specific about the fact that this isn't something the component will just do out of the box—you have to control it.

```suggestion
Use the `open` prop to control the toggling of the Drawer's open and close states, as shown in the demo below:
```

{{"demo": "DrawerBasic.js"}}

## Customization

### Anchor

> Use the anchor prop to specify where the drawer should appear from.

Reviewer: ```suggestion
Use the `anchor` prop to specify which side of the screen the Drawer should originate from.
```

The default value is `left`.

{{"demo": "DrawerAnchor.js"}}

### Size

> Use the size prop to adjust the Drawer's width and height when positioned left/right and top/bottom, respectively.

Reviewer: This is a tricky sentence!

```suggestion
Use the `size` prop to adjust the Drawer's width (when anchored to the left or right) or height (when anchored to the top or bottom).
```

{{"demo": "DrawerSize.js"}}

> ### Close button

Reviewer: let's add an import for the Modal Close component here

> Use the Modal Close component to add a close button to the Drawer.
> It automatically handles the `onClick` event.

Reviewer: We can combine these two sentences into one

```suggestion
Use the Modal Close component to add a close button to the Drawer that automatically handles the `onClick` event.
```

{{"demo": "DrawerCloseButton.js"}}

### Transition

> Customize the Drawer's opening transition by using the CSS variables below inside the sx prop:

Reviewer: ```suggestion
Customize the Drawer's opening transition by using the CSS variables below inside the `sx` prop:
```

Reviewer: Maybe we should link to the sx prop doc page here too? Can't necessarily assume that any casual user knows what it is

- `--Drawer-transitionFunction`: the [transition function](https://developer.mozilla.org/en-US/docs/Web/CSS/transition-timing-function); default is `ease`.
- `--Drawer-transitionDuration`: the [duration of the transition](https://developer.mozilla.org/en-US/docs/Web/CSS/transition-duration); default is `0.3s`.

{{"demo": "DrawerTransition.js"}}

> ### Scrollable content

Reviewer: Dialog Content import here

Use the Dialog Content component to create a scrollable container inside the Drawer.

{{"demo": "DrawerScrollable.js"}}

## Common examples

### Mobile navigation

> A common use case for the Drawer is mobile menus.

Reviewer: ```suggestion
A common use case for the Drawer is to build mobile-friendly navigation menus:
```

{{"demo": "DrawerMobileNavigation.js"}}

### Inset drawer

> To make the Drawer not glued to the edge of the viewport, creating an inset style, use a combination of background and padding values on the Drawer component and either the Box or [Sheet](/joy-ui/react-sheet/) components for the full-height wrapper.

Reviewer: Another tricky sentence—in this case I think it's best to extract the description of an inset drawer from the steps necessary to build one.

```suggestion
An inset drawer is a panel that's not anchored to any side of the screen.
You can achieve this by applying background and padding values to the Drawer component, and using either the Box or [Sheet](/joy-ui/react-sheet/) components for a full-height wrapper, as shown below:
```
